node/src/components/storage.rs

```diff
@@ -1474,29 +1557,18 @@ impl Storage {
             .collect()
     }
 
-    /// Get the switch block for a specified era number in a read-only LMDB database transaction.
-    ///
-    /// # Panics
-    ///
-    /// Panics on any IO or db corruption error.
-    pub fn transactional_get_switch_block_by_era_id(
+    /// Get the switch block for a specified era number.
+    pub fn read_switch_block_by_era_num(
         &self,
         switch_block_era_num: u64,
-    ) -> Option<Block> {
-        let mut read_only_lmdb_transaction = self
-            .env()
-            .begin_ro_txn()
-            .expect("Could not start read only transaction for lmdb");
-        let switch_block = self
-            .get_switch_block_by_era_id(
-                &mut read_only_lmdb_transaction,
-                EraId::from(switch_block_era_num),
-            )
-            .expect("LMDB panicked trying to get switch block");
-        read_only_lmdb_transaction
-            .commit()
-            .expect("Could not commit transaction");
-        switch_block
+    ) -> Result<Option<Block>, Error> {
+        let mut read_only_lmdb_transaction = self.env().begin_ro_txn()?;
+        let maybe_switch_block = self.get_switch_block_by_era_id(
+            &mut read_only_lmdb_transaction,
+            EraId::from(switch_block_era_num),
+        )?;
+        drop(read_only_lmdb_transaction);
```

Comment:
I think this line could just be deleted.

Reply:
Fixed in e669360a7